aiohttp/abc.py

```diff
@@ -146,3 +146,27 @@ def write_eof(self, chunk=b''):
     @abstractmethod
     def drain(self):
         """Flush the write buffer."""
+
+
+class AbstractAccessLogger(ABC):
+
+    def __init__(self, logger, log_format):
+        self.logger = logger
+        self.log_format = log_format
+
+    @abstractmethod
+    def _log(self, request, response, time):
+        """
+
+        :param request: Request object.
+        :param response: Response object.
+        :param float time: Time taken to serve the request.
+        """
+
+    def log(self, request, response, time):
+        """Emit log to logger"""
+
+        try:
+            self._log(request, response, time)
```

Comment:
Let's drop `_log()` and make `log()` abstract method.
On one hand formatting logic always should be wrapped by catching exception obviously.
On other -- let's not make any decision about implementation details.
Implementor might want to do other things than just registering caught exception.
Let's keep ABC very abstract, my guts feel that there are many contradictory might occur if we will bake a code in abstract class.